gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -160,13 +165,22 @@ private void addSortLimitAndOffset(
     }
 
     // handle group by scenario with group limit set
-    if (requestContext.hasGroupBy() && request.getGroupLimit() > 0) {
-      // in group by scenario, set limit to minimum of limit or group-limit
-      queryBuilder.setLimit(Math.min(request.getLimit(), request.getGroupLimit()));
+    if (requestContext.hasGroupBy()) {
+      int limit = request.getLimit();
+      if (request.getGroupLimit() > 0) {
+        // in group by scenario, set limit to minimum of limit or group-limit
+        limit = Math.min(request.getLimit(), request.getGroupLimit());
+      }
+      // pinot doesn't handle offset with group by correctly
+      // we will add offset to limit itself and then ignore results till offset in response
+      limit += request.getOffset();
+      // don't exceed default group by limit
```

Comment:
Can you throw in this case? Better an error than incorrect results

Reply:
dome